wrangler-core/src/main/java/io/cdap/directives/row/RecordConditionFilter.java

```diff
@@ -129,4 +134,16 @@ public List<EntityCountMetric> getCountMetrics() {
     EntityCountMetric jexlCategoryMetric = getJexlCategoryMetric(el.getScriptParsedText());
     return (jexlCategoryMetric == null) ? null : ImmutableList.of(jexlCategoryMetric);
   }
+
+  @Override
+  public Relation transform(RelationalTranformContext relationalTranformContext,
+                            Relation relation) {
+    java.util.Optional<ExpressionFactory<String>> expressionFactory = SqlExpressionGenerator
+            .getExpressionFactory(relationalTranformContext);
+    if (!expressionFactory.isPresent()) {
+      return new InvalidRelation("Cannot find an Expression Factory");
+    }
+    return relation.filter(expressionFactory.get().compile(el.getScriptParsedText()));
```

Comment:
Add a code comment/TODO stating that potentially some JEXL expressions (not supported in SQL) can break this. Same in ColumnExpression and other places